Configuration.mk

```diff
@@ -11,6 +11,7 @@ MAKEFLAGS += -r
 MAKEFLAGS += -R
 
 # Toolchain programs
+ifndef CC
```

Comment:
If you have `CC` defined in your own environment, would it define it as `-gcc`, `-llvm`, etc.? I want to make sure we were using it the same way people's environment variables would.

Reply:
I'm not sure what you mean.

My `CC` is set as: `riscv32-oe-elf-gcc -mabi=ilp32 --sysroot=/usr/local/oecore-x86_64/sysroots/riscv32-oe-elf`

Reply:
If you look, we currently define `CC` as `-gcc` (https://github.com/tock/libtock-c/blob/0d925bf30c2a68c6143f45372cb2f010fa5fd3c6/Configuration.mk#L17)

And a later step  combines that `CC` with a platform specifier such as "arm-none-eabi" (https://github.com/tock/libtock-c/blob/0d925bf30c2a68c6143f45372cb2f010fa5fd3c6/AppMakefile.mk#L98 ) In this case the platform specifier is `$(2)`.

So, just replacing `CC` with `riscv32-oe-elf-gcc -mabi=ilp32 --sysroot=/usr/local/oecore-x86_64/sysroots/riscv32-oe-elf` is not how the build system is set up to work right now for libtock-c. (It might still technically work, because it'll replace the platform prefix with nothing depending on the input architecture).

Reply:
An argument could be made that we are using `CC` _wrong_ and should fix it. But I think just replacing it with the user's environment variable is not currently the correct choice.

Reply:
Also, it's come up in calls, but wow I had forgotten how convoluted the libtock-c build system has gotten. I think I introduced the foreach $$ stuff, and I regret it. It's definitely past time to investigate different build systems for libtock-c to support building for multiple architectures with a simpler setup than Make provides.

Reply:
Ok, I'm going to drop the `CC` change and we can revisit in another PR as I think you are right we will need more `Makefile` work.